### ISSUE TITLE
Completed episodes sometimes stuck with progress in jellyfin

### ISSUE THREAD

**Issue by tadly** [jellyfin/jellyfin-kodi#538]
**Describe the bug**
IIRC this started when I migrated to Kodi 19 back when it was released.

The problem is that every now and then, after finishing an episode of something, the next day it would show up in kodi/jellyfin
as "continue watching" with some random progress left.
The reason I say "next day" is because at the time of finishing it, I see it disappear from the listing within kodi (I have it filtered to only show unwatched).
This could be because kodi updates the entry in the db and some time later (or the next day) jellyfin updates it again with its state.

Yesterday it happend for 2/3 of the episodes I watched
* Episode 11 finished at 20:27 (properly marked as watched)
* Episode 12 finished at 20:49 (stuck with progress left)
* Episode 13 finished at 21:15 (stuck with progress left)

**To Reproduce**
I have no way of reproducing it as it happens seemingly random.

**Expected behavior**
Episodes should me marked as watched

**Logs**
Kodi log: https://paste.ubuntu.com/p/C96JV28wHD/
Jellyfin log: https://paste.ubuntu.com/p/tkxmTJ4qxd/

**System (please complete the following information):**
 - OS: Linux
 - Jellyfin Version: 10.7.5
 - Kodi Version: 19.1
 - Addon Version: 0.7.4+py3
 - Playback Mode: Native

**Additional context**

**Comment by tadly** [comment]
Can really no one look into this?

Even after a full kod reinstall this is still a ongoing issue

**Comment by mcarlton00** [comment]
We're all volunteers and on limited time, so things don't always get fixed quickly.  Adding on to that, this is a super niche issue that we've never seen before, and the errors in your logs don't make any sense at all.  To top it off, you're using native mode, which nobody on the team uses (or even likes existing, to be honest).  The only thing I can imagine that even remotely makes sense is an unstable internet connection, so playback reporting to the server is hit or miss.  The fact that the logs keep puking on parsing the response from api calls seems to support this, but again, it's not an error we've ever seen or been able to replicate.  The error that's actually being thrown is deep within one of our dependencies, which is super unusual and seems to indicate that you're receiving corrupt json from the server (which shouldn't be possible).

TLDR: Nobody else has experienced this or been able to recreate it, and without that we can't really do anything to fix it.  And with that in mind, this is likely going to be a time suck to debug that none of us (mainly me, at this point) have time for right now.  We'll circle back around to it sooner or later, but we can only do so much with the time we have.

**Comment by tadly** [comment]
You already help more than you might think with your comment :)  
I would almost bet a buck that this issue stems from native playback mode.
I would prefer to use it because it'll outperform add-on mode any time of the week but I'll give that a try anyways.

I assume the reason you don't like it very much is because you have to do playback monitoring which you "automatically" have with  add-on mode?
Anything else? Really curious as to why it's not liked very much.

**Comment by mcarlton00** [comment]
Mostly the reasons we don't like native mode is because in 95% of use cases, it's entirely unnecessary and is just extra set up.  Addon mode will direct play virtually everything by default.  The only use cases where native mode provides a real benefit that I'm aware of is if your LAN is poo and can't handle the bandwidth, or if your JF server is remote but your media is local.

**Comment by oddstr13** [comment]
I was sure this was a duplicate, but apparently the issue wasn't tracked on GitHub yet.

I can confirm that this is also an (highly intermittent) issue also with addon mode.
Initially the episode is marked as watched in Kodi, but then after a while the playback progress gets synced back from Jellyfin.

I suspect it could be something like the progress not getting updated as 100% but instead *only* marked as watched. I *think* I've only seen this on anime, and suspect it's more likely to happen on shorter episodes.

I haven't looked deeper into this issue, as it only happens some times, and is """solved""" by manually marking the episode as watched in Kodi.

**Comment by tadly** [comment]
I've switched to addon mode and so far I haven't had this issue while with native mode it happend almost every day.

I do watch anime on a daily basis (usually 2 episodes) where only one of the two would end up with some progress left (I think the second one).
I also always skip the outro which might play into this issue?

**Comment by tadly** [comment]
Quick update.

Before I had changed from native to addon mode, kodi would crash **multiple** times a day (usually 3-5 times) and it didn't matter what I was doing.
I could have been listening to music with a different addon or playing a youtube video (aka nothing jellyfin related).

Ever since I changed to addon mode, kodi stopped crashing altogether. Though I'd mention it...

**Comment by Ulrar** [comment]
Same issue here. I also feel like the issue is the server sending back a progress update after the episode completed, overriding Kodi's local completion. I've noticed that opening the episode and quickly seeking to the end does not work, to clear it I have to seek close to the end and let normal play back finish the episode itself, and even that sometimes does not clear it.
But that only happens if an episode had a progress marker (was stopped and resumed), as far as I know it never happened with an episode that was just played all the way through on the first time.


As a side note since it came up above, I use native mode for three reasons : 
- it feels a lot snappier when seeking
- it lets Kodi find the EDL files generated by intro skipper. In addon mode intro skipper is entirely gone, unless I enable auto skipping server side, which I don't like.
- subtitles always work. It pretty much refuses to select the track I want by default regardless of settings, but when I select it manually it works. In addon mode it's a bit hit or miss, which isn't surprising since it's also unreliable in the jellyfin app itself.

I personally love native mode, and have been half considering setting up VPNs over the internet to let everyone switch to it.

**Comment by sic6SaNdMaN** [comment]
From #606:

I can reproduce this behaviour on all my devices and Kodi installations and it's going on for at least a year now. I hoped, it will get fixed.
And it didn't bother me that much, because normally I watch an episode nearly to it's end, so it will be recognized as "watched".
But now I have several series, where there are 5 minutes or more of endcredits and I skip them, of course.
So every episode stays as "partially watched".

And that's the important point: Basis is a "partially watched" episode or film.
When I now use "mark as watched" in Kodi, this doesn't get synced to the Jellyfin server.
Same goes for "reset position" (I don't know the exact english label for it), that doesn't reach the server either.

The only thing that works and gets synced is

first "mark as unwatched"
then "mark as watched" again.
This is my workaround and has been for a long time.
But doing this on every episode is kinda annoying...
